R/database.R

```diff
@@ -1432,40 +1488,59 @@ get_ssm_by_regions = function(regions_list,
   if(missing(maf_data)){
     region_mafs = lapply(regions, function(x){get_ssm_by_region(region = x,
                                                                 streamlined = streamlined,
+                                                                basic_columns = TRUE,
                                                                 from_indexed_flatfile = from_indexed_flatfile,
                                                                 mode = mode,
                                                                 augmented = augmented,
                                                                 seq_type = seq_type,
-                                                                projection = projection,
-                                                                allow_clustered = allow_clustered)})
+                                                                projection = projection)})
   }else{
     region_mafs = lapply(regions, function(x){get_ssm_by_region(region = x,
                                                                 streamlined = streamlined,
                                                                 maf_data = maf_data,
                                                                 from_indexed_flatfile = from_indexed_flatfile,
-                                                                mode = mode,
-                                                                allow_clustered = allow_clustered)})
+                                                                mode = mode)})
   }
+
   if(!use_name_column){
     rn = regions
   }else{
     rn = regions_bed[["name"]]
   }
+
+  #get region names
   tibbled_data = tibble(region_mafs, region_name = rn)
   unnested_df = tibbled_data %>%
     unnest_longer(region_mafs)
 
+  #subset on region names and rename column before cbind with full maf
+  r_names = as.data.frame(unnested_df$region_name)
+  colnames(r_names)[1] <- "region_name"
+
+  #unnest maf
+  unnested_maf = bind_rows(region_mafs)
+
+  #join region names df with unnested maf
+  maf = cbind(r_names, unnested_maf)
+
   if(streamlined){
-    unlisted_df = mutate(unnested_df, start = region_mafs$Start_Position, sample_id = region_mafs$Tumor_Sample_Barcode) %>%
```

Comment:
I'm very confused by this change and how it maintains the original functionality

Reply:
There might be a more elegant solution to this. My update achieves the same output in the following steps:

1. Subsets region names from the unnested df
2. Renames the column (to region_name, to match downstream required format)
3. Binding maf df (unnests)
4. Joins the df with region names with unnested maf. 
 
The integrity of the final returned maf is retained (at least to the testing I've done). Does this clarify the confusion?